opm/autodiff/RateConverter.hpp

```diff
@@ -689,20 +691,22 @@ namespace Opm {
                     voidage_rates[iw] = surface_rates[iw] / bw;
                 }
 
+                // Use average Rs and Rv:
+                double Rs = std::min(ra.rs, surface_rates[ig]/(surface_rates[io]+1.0e-15));
+                double Rv = std::min(ra.rv, surface_rates[io]/(surface_rates[ig]+1.0e-15));
+
```

Comment:
@GitPaean This is the change you should look at.

Reply:
It provides some guard for the un-reasonable rs and rv values for specific well rates composition.  Will come back later with more thoughts.

Reply:
Without this fix, basically, with the formulations below, there will be possibly some gas/oil reservoir volume rates can be negative (if it did not happen, then it means the above fix did not take effects at all.). But for under-saturated situation, gas is dissolved in the oil completely, there will be no gas phase. Under this situation, will there be any oil vaporized into gas? I guess the reasonable assumption will be no.  Then it means rv should be zero. Since rv here we are using are from field/region average, it typically is not zero (we should check this), which means our formulations need to change a little bit to consider the under-saturated situation.  We should design an under-saturated case to verify our guess and test/design the algorithms. Big models can have other distractions, which makes the verification more difficult.

Reply:
> We should design an under-saturated case to verify our guess and test/design the algorithms.

 I agree.